src/loginPage/loginPage.component.tsx

```diff
@@ -551,6 +553,10 @@ const LoginPageComponent = (props: CombinedLoginProps): React.ReactElement => {
         {props.auth.loading ? (
           <CircularProgress className={props.classes.spinner} />
         ) : null}
+
+        {typeof mnemonic !== 'undefined' && !fetchedMnemonics ? (
+          <CircularProgress className={props.classes.spinner} />
+        ) : null}
```

Comment:
These could still be incorporated into one conditional - I don't know if that's more or less readable

Reply:
It's still readable. I have added the changes